Upstream-Status: Pending
---

diff --git a/boards/mcimx95evk/sm/brd_sm_handlers.c b/boards/mcimx95evk/sm/brd_sm_handlers.c
index a96f6ce..1a06780 100755
--- a/boards/mcimx95evk/sm/brd_sm_handlers.c
+++ b/boards/mcimx95evk/sm/brd_sm_handlers.c
@@ -96,6 +96,7 @@ int32_t BRD_SM_SerialDevicesInit(void)
 {
     int32_t status = SM_ERR_SUCCESS;
     LPI2C_Type *const s_i2cBases[] = LPI2C_BASE_PTRS;
+#if 0
     pcal6408a_config_t pcal6408Config;
 
     /* Fill in PCAL6408A dev */
@@ -116,6 +117,7 @@ int32_t BRD_SM_SerialDevicesInit(void)
             status = SM_ERR_HARDWARE_ERROR;
         }
     }
+#endif
 
     if (status == SM_ERR_SUCCESS)
     {
@@ -306,11 +308,12 @@ int32_t BRD_SM_BusExpMaskSet(uint8_t val, uint8_t mask)
 void GPIO1_0_IRQHandler(void)
 {
     uint32_t flags;
-    uint8_t status, val;
+    //uint8_t status, val;
 
     /* Get GPIO status */
     flags = RGPIO_GetPinsInterruptFlags(GPIO1, kRGPIO_InterruptOutput0);
 
+#if 0
     /* Get PCAL6408A status */
     (void) PCAL6408A_IntStatusGet(&g_pcal6408aDev, &status);
 
@@ -343,6 +346,15 @@ void GPIO1_0_IRQHandler(void)
 
     /* Handle controls interrupts */
     BRD_SM_ControlHandler(status, val);
+#else
+    /* Clear GPIO interrupts */
+    RGPIO_ClearPinsInterruptFlags(GPIO1, kRGPIO_InterruptOutput0, flags);
+
+    BRD_SM_Pf09Handler();
+
+    if (g_pca2131Used)
+        BRD_SM_BbmHandler();
+#endif
 
     /* Adjust dynamic IRQ priority */
     (void) DEV_SM_IrqPrioUpdate();
