From b31dd0dcd28605df8e08904d5c8b44de4a16ccd8 Mon Sep 17 00:00:00 2001
From: "tony.shih" <tony13.shih@advantech.com>
Date: Fri, 25 Jul 2025 14:26:31 +0800
Subject: [PATCH] add-iomux-swrst

Upstream-Status: Inappropriate [Board-specific change for aom5521]
---
 board/freescale/imx95_evk/imx95_evk.c | 28 +++++++++++++++++++++++++++
 1 file changed, 28 insertions(+)

diff --git a/board/freescale/imx95_evk/imx95_evk.c b/board/freescale/imx95_evk/imx95_evk.c
index 9790c610ac7..347b2c7cff8 100644
--- a/board/freescale/imx95_evk/imx95_evk.c
+++ b/board/freescale/imx95_evk/imx95_evk.c
@@ -45,6 +45,32 @@ struct efi_capsule_update_info update_info = {
 };
 #endif /* EFI_HAVE_CAPSULE_SUPPORT */
 
+static void setup_iomux_swrst(void)
+{
+	int ret;
+	struct gpio_desc desc;
+
+	printf("[ADVANTECH] setup_iomux_swrst\n");
+
+	ret = dm_gpio_lookup_name("GPIO2_27", &desc);
+	if (ret) {
+		printf("%s lookup GPIO2_27 failed ret = %d\n", __func__, ret);
+		return;
+	}
+
+	ret = dm_gpio_request(&desc, "SW_RESET_OUT");
+	if (ret) {
+		printf("%s request SW_RESET_OUT failed ret = %d\n", __func__, ret);
+		return;
+	}
+
+	/* assert the SW_RESET_OUT */
+	dm_gpio_set_dir_flags(&desc, GPIOD_IS_OUT | GPIOD_IS_OUT_ACTIVE | GPIOD_ACTIVE_LOW);
+	udelay(10000);
+	dm_gpio_set_value(&desc, 0); /* deassert the SW_RESET_OUT */
+	udelay(80000);
+}
+
 int board_early_init_f(void)
 {
 	/* UART1: A55, UART2: M33, UART3: M7 */
@@ -473,6 +499,8 @@ int board_init(void)
 	netc_regulator_enable("regulator-m2-pwr", true);
 #endif
 
+	setup_iomux_swrst();
+
 	netc_init();
 
 	flexspi_nor_steup();
-- 
2.34.1

